README.md

```diff
@@ -0,0 +1,71 @@
+# HOAG - SHINE
+
+This is the first part of the code for the paper "SHINE: SHaring the INverse Estimate from the forward pass for bi-level optimization and implicit models", submitted at the 2021 NeurIPS conference.
+This source code allows to reproduce the experiments on logistic regression, i.e. Figures 1-3, and Figure E.1. in Appendix.
+
+## General instructions
+
+You need Python 3.7 or above to run this code.
+You can then install the requirements with: `pip install -r requirements.txt`.
+
+When running the scripts, you will see the following warning printed: `CG did not converge to the desired precision`.
+It does not indicate that there is a problem with your current run.
+
+## Reproducing Figure 1, Quality of the inversion using OPA
+
+Figure 1. can be reproduced by running the `lbfgs_inversion.py` script:
+
+```
+python lbfgs_inversion.py
+```
+
+It will take you about 8 seconds to run this script.
+
+## Reproducing Figure 2, Bi-level optimization
+
+Figure 2. can be reproduced by running the `main_plots.py` script:
+
+```
+python main_plots.py
+```
+
+By default, the results will be re-computed and saved each time.
+If you want to use the results saved from a previous run, you can change the variable `reload_results` to `False` in the script.
```

Comment:
```suggestion
If you want to use the results saved from a previous run, you can change the variable `reload_results` to `True` in the script.
```
No? Else, maybe change to `recompute_results`?

Reply:
No, sure I will change the name

Reply:
done